src/redprl/refiner.fun

```diff
@@ -380,39 +369,39 @@ struct
         val specTy' = substMetaenv rho specTy
         val _ = if Hyps.isEmpty H' then () else raise Fail "Equality.Custom only works with empty sequent"
 
-        val goalTy = makeEqTypeIfDifferentOrLess (I, H) ((ty, specTy'), k) specK
+        val goalTy = makeEqTypeIfDifferentOrNotSubUniv (I, H) ((ty, specTy'), l, k) (specL, specK)
       in
         |>:? goalTy #> (I, H, trivial)
       end
 
     fun Symmetry _ jdg =
       let
         val _ = RedPrlLog.trace "Equality.Symmetry"
-        val (I, H) >> CJ.EQ ((m, n), (ty, k)) = jdg
-        val goal = makeEq (I, H) ((n, m), (ty, k))
+        val (I, H) >> CJ.EQ ((m, n), (ty, l, k)) = jdg
+        val goal = makeEq (I, H) ((n, m), (ty, l, k))
       in
         |>: goal #> (I, H, trivial)
       end
 
     fun RewriteTrue z alpha jdg =
       let
         val _ = RedPrlLog.trace "Equality.RewriteTrue"
-        val (I, H) >> CJ.TRUE (mainGoal, k) = jdg
-        val CJ.EQ ((m, n), (ty, k')) = Hyps.lookup z H
+        val (I, H) >> CJ.TRUE (mainGoal, l, k) = jdg
+        val CJ.EQ ((m, n), (ty, l', k')) = Hyps.lookup z H
 
         val x = alpha 0
-        val Hx = H @> (x, CJ.TRUE (ty, k'))
+        val Hx = H @> (x, CJ.TRUE (ty, l', k'))
         val (motiveGoal, motiveHole) = makeTerm (I, Hx) O.EXP
+        val motiveWfGoal = makeType (I, Hx) (motiveHole, NONE, K.top)
 
         val motiven = substVar (n, x) motiveHole
         val motivem = substVar (m, x) motiveHole
 
-        val (rewrittenGoal, rewrittenHole) = makeTrue (I, H) (motiven, K.top)
-
-        val motiveWfGoal = makeType (I, Hx) (motiveHole, K.top)
-        val motiveMatchesMainGoal = makeEqType (I, H) ((motivem, mainGoal), k)
+        val (rewrittenGoal, rewrittenHole) = makeTrue (I, H) (motiven, l, k)
+        (* favonia: this should be enough... is it? *)
+        val motiveMatchesMainGoal = makeEqTypeIfDifferent (I, H) ((motivem, mainGoal), l, k)
```

Comment:
The thing is that I'm not 300% sure if levels and kindhood all (eventually at the omega level) respect equality. They should, and if they don't, I think it's a bug. @cangiuli